draft-ietf-jsonpath-base.md

```diff
@@ -1449,13 +1449,23 @@ Queries:
 
 ## Normalized Paths
 
-A Normalized Path is a JSONPath with restricted syntax that identifies a node by providing a query that results in exactly that node. For example,
-the JSONPath expression `$.book[?(@.price<10)]` could select two values with Normalized Paths
-`$['book'][3]` and `$['book'][5]`. For a given JSON value, there is a one to one correspondence between the value's
+A Normalized Path is a JSONPath with restricted syntax that identifies a node by providing a query that results in exactly that node,
+for example, `$['book'][3]`.
+
+A Normalized Path provides a concrete string representation of a node in a nodelist.
+Consequently, a JSONPath implementation may output Normalized Paths instead of, or in addition to, the values identified by these paths.
+For example, the JSONPath expression `$.book[?(@.price<10)]` could select two values with Normalized Paths
+`$['book'][3]` and `$['book'][5]`.
+
+Normalized Paths are used in this document to clarify the contents of nodelists in examples.
+
+Since output nodelists may be input to subsequent processing (e.g., removal of duplicate nodes), it is essential to provide a standard
+string representation of their nodes.
+
+For a given JSON value, there is a one to one correspondence between the value's
```

Comment:
Should be 
```
For a given JSON value, there is a one to one correspondence between the value's
node and the Normalized Path that identify this node.
```

Reply:
one-to-one with hyphens?

Reply:
> Should be
> 
> ```
> For a given JSON value, there is a one to one correspondence between the value's
> node and the Normalized Path that identifies this node.
> ```

That wasn't what I was trying to say. The idea is that all the nodes in the value, including descendants, are in a one-to-one correspondence with their Normalized Paths. Please see 1538578f1cd9df6de27c9eae9d2da09f6cb5dee9 for a rewording.

Reply:
Hmm ... if it is true for any single value and its associated normalized Path, it is consequently true for all values.

So the singular formulation above is sufficient IMHO.

Reply:
A one-to-one correspondence, also known as a bijection, is only really useful when there is more than one item in the domain and range of the correspondence. There is always a one-to-one correspondence between one item and another taken in isolation. So the singular formulation would essentially say nothing.

This is a little subtle because the one-to-one correspondence is a consequence of the way we have defined Normalized Paths, but I want to point it out since it's a very useful property.

To see the general point, let's step away from Normalized Paths briefly to a fictional "label" of a node and look at an example value `[0]` with an array node and a number node. Taking these nodes individually, we could say that `"x"` is the label of the array node and that `"x"` is also the label of the number node. Then:
* There is a one-to-one correspondence between the array node and the label `"x"`.
* There is a one-to-one correspondence between the number node and the label `"x"`.
* But there isn't a one-to-one correspondence between the set of nodes and their labels, since both nodes have the same label.

Reply:
thanks for enlightening me ...

Reply:
You're welcome. However, given the complexity of the one-to-one correspondence language, I took the opportunity to simplify things by talking about a Normalized Path being an identity of a node. This implies a one-to-one correspondence, but I think it's easier for readers to understand.